interview-questions.md

```diff
@@ -4,43 +4,79 @@ Answer the following questions. First, without external resources. Challenge you
 1. In a model called Animal that has_many Sightings, what is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
   Your answer:
-
+  The foreign_key would live inside the Sightings(belongs_to Animal) model as a column. 
+  The foreign_key would be "animal_id"
   Researched answer:
-
+  Didnt really do an on the spot research here. But relational models work somewhat similar to the parent-child hiarchy- where the parent would be Animal, which would be defined as a "has_many" of the child Sightings which "belongs_to" Animal(parent). The foreign_key would need to belong inside of the child model so that it can relate back to its parent model, Animal. The general structure of the foreign_key would be "parent_id", so in this instance we would have it as "animal_id"
 
 
 2. Which routes must always be passed params and why?
 
   Your answer:
+  Show - Requires an id param to be able to pick out the specific row/entry you are looking for. 
+  Update - Requires the id param to be able to pick out the specific row/entry you are looking to update/modify
+  Delete - Requires the id param to be able to pick out the specific row/entry you are looking to DESTROY (muahaha.)
+  
+  And honestly, if you're doing anything somewhat custom in your controller, you might be needing to create a route that will require a param passed to it, but the shown examples above are just three of the basics I can think of at this time when using something like "generate resource" that creates a set of defaults for you.  
 
   Researched answer:
+  This would be a huge wall of text, if I were to copy paste directly in, so I'm not going to do that. However the best resource is directly from Rails guide documentation itself. So heres a link! https://guides.rubyonrails.org/routing.html
 
 
 
 3. Write a rails route for a controller called "main" and a page called "game" that takes in a parameter called "guess".
 
   Your answer:
+  In the situation of a interview, I'd be asking a few more questions- such as what the model name is and what is goin' on in the controller. So, to the best of my abilities I'm working only with what I have here. 
+get '/game/:guess', to: 'main#game'
+  
```

Comment:
looks good to me! great thinking about what further questions you'd ask in an interview